libthymesisflow/src/thymesisflow.c

```diff
@@ -0,0 +1,243 @@
+/// @file thymesisflow.c
+#include "thymesisflow.h"
+#include <fcntl.h>
+
+int detach(const uint64_t memory_size) {
```

Comment:
I would consider renaming this method to something more descriptive. The name `detach` could be too generic for a code base dealing with attaching and detaching memory.